vault/logical_system.go

```diff
@@ -2245,6 +2269,47 @@ func (b *SystemBackend) handleTuneWriteCommon(ctx context.Context, path string,
 		}
 	}
 
+	if rawVal, ok := data.GetOk("identity_token_key"); ok {
+		identityTokenKey := rawVal.(string)
+
+		// Ensure the key exists in the identity store
+		if identityTokenKey != "" {
+			identityStore := b.Core.IdentityStore()
```

Comment:
should we move the validation code to ensure the key exists to a common util?

Reply:
Good call! I'll do that.

Reply:
It ended up being not much of an improvement to create a helper function. Also, we need to hold the identity store read lock until the mount/tune is complete so that we don't race with the deletion handler in the identity store. I ended up switching the approach in https://github.com/hashicorp/vault/pull/24962/commits/5a25a2527bf35fe576a146e5668d563cc0c9c9a4.

Reply:
Ah gotcha, makes sense. Thanks for looking into it!